moengage-segment-kotlin-destination/build.gradle.kts

```diff
@@ -0,0 +1,43 @@
+plugins {
+    id("org.jetbrains.kotlin.plugin.serialization")
+    alias(moengageInternal.plugins.plugin.dokka)
+    alias(moengageInternal.plugins.plugin.android.lib)
+    alias(moengageInternal.plugins.plugin.kotlin.android)
+}
+
+apply(from = "../scripts/gradle/release.gradle")
+
+val libVersionName = project.findProperty("VERSION_NAME") as String
+
+android {
+    compileSdk = 31
+
+    defaultConfig {
+        minSdk = 21
+        targetSdk = 31
+
+        buildConfigField("String", "MOENGAGE_SEGMENT_KOTLIN_VERSION", "\"$libVersionName\"")
+    }
+
+    buildTypes {
+        release {
+            isMinifyEnabled = false
+            proguardFiles(
+                getDefaultProguardFile("proguard-android-optimize.txt"),
+                "proguard-rules.pro"
+            )
+        }
+    }
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_1_8
```

Comment:
According to the docs, the segment SDK internally uses a number of Java 8 language APIs through desugaring.
We have to enable desugaring since the minSdk is < 26

Reply:
Done